src/mlpack/tests/load_save_test.cpp

```diff
@@ -2527,3 +2527,140 @@ TEST_CASE("LoadCSVHeaderTest", "[LoadSaveTest]")
   REQUIRE(headers.at(2) == "c");
   REQUIRE(headers.at(3) == "d");
 }
+
+#if ARMA_VERSION_MAJOR >= 12
```

Comment:
```suggestion
// These tests only work with Armadillo 12, as we need the `strict` option to be
// available in Armadillo.
#if ARMA_VERSION_MAJOR >= 12
```
Just a suggestion, it would be helpful to comment why we can only run these tests in some cases.